active/003-seq-feature.md

```diff
@@ -0,0 +1,180 @@
+
+   ```python
+>>> from pandas.core.interval import Interval, IntervalIndex
+>>> intervals = [(0, 100), (0, 30), (70, 100), (50, 200), (120, 150), (120, 150)]
+>>> iix = IntervalIndex.from_tuples(intervals, closed='left')
+>>> iix
+IntervalIndex(left=[0, 0, 70, 50, 120, 120],
+              right=[100, 30, 100, 200, 150, 150],
+              closed='left')
+>>> gene1 = dict(name='gen1',
+...                 gene_synonym="ECK0007; JW0006")
+>>> cds1 = dict(name='cds1')
+>>> gene2 = dict(name='gen2',
+...                 db_xref=("EcoGene:EG11555", "GeneID:944757"))
+>>> gene3 = dict(name='gen3',
+...                 db_xref="EcoGene:EG11555")
+>>> gene4 = dict(name='gen4')
+>>> interval_md = pd.DataFrame(data=np.array([['gene', 'CDS', 'CDS', 'gene', 'gene', 'gene'],
+...                                  ['+', '+', '+', '+', '+', '+'],
+...                                  [True, True, True, True, True, True],
+...                                  [True, True, True, True, True, False],
+...                                  [gene1, cds1, cds1, gene2, gene3, gene4]]).T,
+...                            copy=True,
+...                            columns=['TYPE', 'STRAND', 'LEFT', 'RIGHT', 'ATTRIBUTES'],
+...                            index=iix)
+>>> interval_md
+            TYPE STRAND  LEFT  RIGHT  \
+[0, 100)    gene      +  True   True
+[0, 30)      CDS      +  True   True
+[70, 100)    CDS      +  True   True
+[50, 200)   gene      +  True   True
+[120, 150)  gene      +  True   True
+[120, 150)  gene      +  True  False
+
+                                                   ATTRIBUTES
+[0, 100)    {'name': 'gen1', 'gene_synonym': 'ECK0007; JW0...
+[0, 30)                                      {'name': 'cds1'}
+[70, 100)                                    {'name': 'cds1'}
+[50, 200)   {'db_xref': ('EcoGene:EG11555', 'GeneID:944757...
+[120, 150)     {'db_xref': 'EcoGene:EG11555', 'name': 'gen3'}
+[120, 150)                                   {'name': 'gen4'}
+   ```
+
+## Types of feature queries
+* query by location. `IntervalIndex` module implements `IntervalTree` for fast query by location. To get features covering a unit or a region, we can do this:
+
+``` python
+>>> interval_md.loc[30, ]
+          TYPE STRAND  LEFT RIGHT  \
+[0, 100)  gene      +  True  True
+
+                                                  ATTIBUTES
+[0, 100)  {'name': 'gen1', 'gene_synonym': 'ECK0007; JW0...
+>>> interval_md.loc[Interval(10, 30), ]
+          TYPE STRAND  LEFT RIGHT  \
+[0, 100)  gene      +  True  True
+[0, 30)    CDS      +  True  True
+
+                                                 ATTRIBUTES
+[0, 100)  {'name': 'gen1', 'gene_synonym': 'ECK0007; JW0...
+[0, 30)                                    {'name': 'cds1'}
+```
+
+* query by attributes. It can be used to fetch multiple features or a single feature. For example:
```

Comment:
Nope, there is no other way to distinct two features if they have exactly the same attributes, which I would argue they shouldn't at the first place if they are 2 features. But I can imagine some tricky cases. Open for other arguments.